src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicRename.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.OutOfProcess;
+using Roslyn.Test.Utilities;
+using Roslyn.VisualStudio.IntegrationTests.Extensions.Editor;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.VisualBasic
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class BasicRename : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.VisualBasic;
+
+        private InlineRenameDialog_OutOfProc InlineRenameDialog => VisualStudio.Instance.InlineRenameDialog;
+
+        public BasicRename(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(BasicRename))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Rename)]
+        public void VerifyLocalVariableRename()
+        {
+            var markup = @"
+Imports System
+Imports System.Collections.Generic
+Imports System.Linq
+
+Module Program
+    Sub Main(args As String())
+        Dim [|x|]$$ As Integer = 0
+        [|x|] = 5
+        TestMethod([|x|])
+    End Sub
+    Sub TestMethod(y As Integer)
+
+    End Sub
+End Module";
+            SetUpEditor(markup);
+            InlineRenameDialog.Invoke();
+
+            MarkupTestFile.GetSpans(markup, out var _, out ImmutableArray<TextSpan> renameSpans);
+            var tags = Editor.GetTagSpans(InlineRenameDialog.ValidRenameTag);
+            AssertEx.SetEqual(renameSpans, tags);
+
+            Editor.SendKeys(VirtualKey.Y, VirtualKey.Enter);
+            this.VerifyTextContains(@"
+Imports System
+Imports System.Collections.Generic
+Imports System.Linq
+
+Module Program
+    Sub Main(args As String())
+        Dim y As Integer = 0
+        y = 5
+        TestMethod(y)
+    End Sub
+    Sub TestMethod(y As Integer)
+
+    End Sub
+End Module");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Rename)]
+        public void VerifyLocalVariableRenameWithCommentsUpdated()
+        {
+            var markup = @"
+Imports System
+Imports System.Collections.Generic
+Imports System.Linq
+
+Module Program
+    ''' <summary>
+    ''' creates a varixable named [|x|] xx
```

Comment:
Intentional, to make sure it doesn't update the x in varixable. Maybe that could use a comment.